### PR TITLE
Add support for clamping for NativeAnimated on iOS

### PR DESCRIPTION
**Pull request by ryangomba**
This diff adds support for clamping on iOS. It separates out code originally submitted in #9048.

Test plan (required)

Run UIExplorer NativeAnimated examples before and after - compare the results. Pay special attention to the new clamped spring example.

### CODE REVIEW

**Comment by facebook-github-bot-4** [comment]
By analyzing the blame information on this pull request, we identified @buba447 and @janicduplessis to be potential reviewers.

**Comment by janicduplessis** [comment]
@ryangomba Could you make the changes I talked about in the other PR? Pretty much just add support for extrapolate types as you mentioned in your TODO. Here's the android PR for reference #9366.

File: Libraries/Animated/src/NativeAnimatedHelper.js [hunk 1]
```diff
@@ -135,6 +135,9 @@ function validateInterpolation(config: Object): void {
   var SUPPORTED_INTERPOLATION_PARAMS = {
     inputRange: true,
     outputRange: true,
+    extrapolate: true,
```
Comment (janicduplessis) [inline Libraries/Animated/src/NativeAnimatedHelper.js]: This won't be needed as it is already added in the android PR.

File: Libraries/NativeAnimation/Nodes/RCTInterpolationAnimatedNode.m [hunk 2]
```diff
@@ -29,6 +31,17 @@ - (instancetype)initWithTag:(NSNumber *)tag
       }
     }
     _outputRange = [outputRange copy];
+
+    if ([config[@"extrapolate"] isEqualToString:@"clamp"]) {
```
Comment (janicduplessis) [inline Libraries/NativeAnimation/Nodes/RCTInterpolationAnimatedNode.m]: JS will only send "extrapolateLeft" and "extrapolateRight" so no need to handle this case.

**Comment by ryangomba** [comment]
I will also follow up on this today

**Comment by ryangomba** [comment]
@janicduplessis this is ready for your review

**Comment by janicduplessis** [comment]
Awesome! Just updated the PR title since the shipit bot seems to strip [TAGS].

@facebook-github-bot shipit

**Pull request closed without merging**